### PR TITLE
chore(discoveryengine): add config to generate apiv1

### PR DESCRIPTION
**Pull request by quartzmo**

### CODE REVIEW

File: internal/postprocessor/config.yaml [hunk 1]
```diff
@@ -44,6 +44,7 @@ modules:
   - datastream
   - deploy
   - dialogflow
+  - discoveryengine
```
Comment (quartzmo) [inline internal/postprocessor/config.yaml:47]: @codyoss Was the omission of the module configuration intentional for `discoveryengine/v1beta`? Assuming not, adding it now.

**Review by noahdietz — COMMENTED** [review]
No need to change the release-please config like in https://github.com/googleapis/google-cloud-go/pull/7987/files ?

File: internal/postprocessor/config.yaml (hunk 1 above)
Comment (codyoss) [reply internal/postprocessor/config.yaml:47]: I don't think so.

**Comment by quartzmo** [comment]
> No need to change the release-please config like in https://github.com/googleapis/google-cloud-go/pull/7987/files ? Or was that done already?

Correct, already added to `release-please-manifest-submodules.json` and `release-please-config-yoshi-submodules.json`. But it was never added to `modules` in `config.yaml`. This PR fixes that, in addition to adding the `v1` client.

**Review by noahdietz — APPROVED** [review]

**Pull request merged**